src/model/implementation/type-base.ts

```diff
@@ -116,13 +154,10 @@ export abstract class TypeBase implements ModelComponent {
 
     @memorize()
     get effectiveModuleSpecification(): EffectiveModuleSpecification {
-        // TODO - determine where the types is used
-        // need to figure out how to do this without infinite recursion
         return new EffectiveModuleSpecification({
-            orCombinedClauses: this.model.modules.map((module) => ({
-                andCombinedModules: [module.name],
-            })),
-        }).simplify();
+            // clauses being undefined is an error state, recover gracefully
+            orCombinedClauses: this.moduleSpecification?.clauses ?? [],
+        });
```

Comment:
it's not that important because you're unlikely to manually write a non-simplified module declaration (I mainly wrote this helper for the cases where we need to and-combine two specifications). But it's better to have anyway. Added the `.simplify()`